app/services/platform-apps/api/modules/scene-transitions.ts

```diff
@@ -0,0 +1,189 @@
+import { apiMethod, EApiPermissions, IApiContext, Module } from './module';
+import {
+  ETransitionType,
+  ITransitionCreateOptions,
+  TransitionsService,
+} from '../../../transitions';
+import { Inject } from '../../../../util/injector';
+import { PlatformAppsService } from '../../index';
+import * as path from 'path';
+import { TObsValue } from '../../../../components/obs/inputs/ObsInput';
+
+type AudioFadeStyle = 'fadeOut' | 'crossFade';
+
+enum ObsAudioFadeStyle {
+  FadeOut = 0,
+  CrossFade = 1,
+}
+
+type TransitionPointType = 'time' | 'frame';
+
+enum ETransitionPointType {
+  Time = 0,
+  Frame = 1,
+}
+
+type StingerTransitionOptions = {
+  /** Transition type **/
+  type: 'stinger';
+  /** The name of the transition **/
+  name: string;
+  /** A relative URL to a video asset inside a Platform app **/
+  url: string;
+  /** How the audio should fade: fade out or crossfade. **/
+  audioFadeStyle?: AudioFadeStyle;
+  /**
+   * If set to true scene, the scene transition won't be editable by
+   * end-users after being created.
+   */
+  shouldLock?: boolean;
+  /** Whether to monitor audio **/
+  shouldMonitorAudio?: boolean;
+  /** A transition point in milliseconds **/
+  transitionPoint?: number;
+  /** Type of transition point, frame or time **/
+  transitionPointType?: TransitionPointType;
+};
+
+type TransitionOptions = StingerTransitionOptions;
+
+/**
+ * This module can be used to create scene transitions.
+ * It's useful for apps to provide both editable and uneditable transitions for the streamer.
+ */
+export class SceneTransitionsModule extends Module {
+  moduleName = 'SceneTransitions';
+
+  permissions = [EApiPermissions.SceneTransitions];
+
+  @Inject() transitionsService: TransitionsService;
+
+  @Inject() platformAppsService: PlatformAppsService;
+
+  /**
+   * Create a scene transition
+   *
+   * Currently, only stinger transitions are supported, as these are the most
+   * useful for customization.
+   *
+   * @param ctx API context
+   * @param options A description of transition options
+   *
+   * @see {TransitionOptions}
+   */
+  @apiMethod()
+  async createTransition(ctx: IApiContext, options: TransitionOptions) {
+    if (options.type === 'stinger') {
+      const appId = ctx.app.id;
+      const path = options.url;
+
+      if (!isVideo(path)) {
+        throw new Error('Invalid file specified, you must provide a video file.');
+      }
+
+      const { shouldLock = false, name, ...settings } = options;
+
+      const transitionOptions: ITransitionCreateOptions = compose(
+        (settings): ITransitionCreateOptions => ({
+          propertiesManagerSettings: {
+            appId,
+            locked: shouldLock,
+          },
+          settings: {
+            ...settings,
+            path: this.platformAppsService.getAssetUrl(appId, path),
+          } as Dictionary<TObsValue>,
+        }),
+        toObsTransitionOptions(stingerTransitionDefaultOptions),
+      )(settings);
```

Comment:
This is hard to grok.  Let's avoid `compose` until the entire team is bought it, understands the behavior, and agrees on its usefulness.